tm2py/controller.py

```diff
@@ -374,43 +374,24 @@ def _calculate_num_processors(self, cpu_processors: int):
     
     def warmstart_check(self):
         """
-        Check if required input files and folders exist before running the components.
+        Check if required warmstart demand files and folders exist before running the components.
         """
-        _run_components = set(self.config.run.initial_components) | set(self.config.run.global_iteration_components) | set(self.config.run.final_components)
+        _run_components = [comp[1] for comp in self._queued_components]
```

Comment:
Is there an alternative to `comp[1]` so that it's more generic and interpretable? Is it hard to review a hard-coded index.

Reply:
Updated to spell out the elements of the tuple: ``_run_components = [name for iteration, name, component in self._queued_components]``